DashWallet/Sources/UI/CrowdNode/Online/CrowdNodeWebViewController.swift

```diff
@@ -65,5 +76,51 @@ class CrowdNodeWebViewController: UIViewController {
                 }
             }
             .store(in: &cancellableBag)
+        
+        viewModel.$onlineAccountState
+            .removeDuplicates()
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] state in
+                if state == .done {
+                    // TODO: check
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            }
+            .store(in: &cancellableBag)
+        
+        viewModel.$error
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] error in
+                if error != nil {
+                    // TODO: check
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            }
+            .store(in: &cancellableBag)
+    }
+}
+
+extension CrowdNodeWebViewController: WKNavigationDelegate {
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, preferences: WKWebpagePreferences) async -> (WKNavigationActionPolicy, WKWebpagePreferences) {
+
+        if let email = email, let url = navigationAction.request.url?.absoluteString {
+            let fullSuffix = "\(kSignupSuffix)&loginHint=\(email)"
+            
+            if (url.hasPrefix(loginPrefix) && !url.hasSuffix(fullSuffix) && !url.hasSuffix(kCallbackSuffix)) {
+                let redirectUrl = "\(url)\(fullSuffix)"
+                let urlRequest = URLRequest(url: URL(string: redirectUrl)!)
+                webView.load(urlRequest)
+                previousUrl = url
+                
+                return (WKNavigationActionPolicy.cancel, preferences)
+            } else if (previousUrl.hasPrefix(loginPrefix) && url.hasPrefix(accountPrefix)) {
+                // Successful signup
+                viewModel.finishSignUpToOnlineAccount()
+            }
+            
+            previousUrl = url
+        }
```

Comment:
We have to manually figure out when the webview transitions to the profile screen to detect successful signup.